File: js/bon-appetit.js
function bonAppetit(bill, k, b) {
    const annaOwes = (bill.reduce((a, b) => (a + b)) - bill[k]) / 2;
    if (annaOwes === b) {
        console.log(`Bon Appetit`);
    } else {
        console.log(b - annaOwes);
    }

}
bonAppetit([3, 10, 2, 9], 1, 12);

File: src/main/java/problems/BonAppetit.java
package problems;

import java.util.List;

public class BonAppetit {
    static void bonAppetit(List<Integer> bill, int k, int b) {
        int totalBill = bill.stream().mapToInt(i -> i.intValue()).sum();
        int annaOwes = ((totalBill - bill.get(k)) / 2);
        if (annaOwes == b) {
            System.out.println("Bon Appetit");
        } else {
            System.out.println(b - annaOwes);
        }

    }
}
